### PR TITLE
LPS-117531

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang, the pr is based on tinatian#1972. I merge SF (about PortletHotDeployListener class) to the previous commit. For the [commit](https://github.com/dantewang/liferay-portal/commit/2fd26012666597fded7a5f9a36e321959e52d84a), I edit the commit info and keep it because it can simply the logic. For the last commit, it is not related with the fix (the commit included PortletTracker changes. I merge PortletTracker SF to the previous commit, so it isn't related now), I think we may remove it.

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by yuhai** [comment]
ci:test

**Comment by dantewang** [comment]
Fetched

File: portal-impl/src/com/liferay/portal/security/permission/ResourceActionsImpl.java [hunk 1]
```diff
@@ -1076,14 +1080,23 @@ private void _read(
...
+				else {
+					portlet = portletLocalService.getPortletById(portletName);
+				}
 
 				Set<String> portletActions = _getPortletMimeTypeActions(
 					portletName, portlet);
 
 				if (!portletName.equals(PortletKeys.PORTAL)) {
-					_checkPortletActions(portlet, portletActions);
+					_checkPortletLayoutManagerActions(portletActions);
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/security/permission/ResourceActionsImpl.java]: @yuhai What was the reason we changed this?

**Comment by dantewang** [comment]
Pull request submitted to https://github.com/dantewang/liferay-portal/pull/1750. [See changes here.](https://github.com/dantewang/liferay-portal/compare/ca475be2301a~3...ca475be2301a)

**Pull request closed without merging**